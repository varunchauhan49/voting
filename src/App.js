import React, { Component } from 'react';
import Poll from './Poll';
import PollList from './PollList';
import { Card, CardBody, CardColumns, CardHeader, Row, Col } from 'reactstrap';
import pollData from './pollData';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';

// defaults.labelFontSize = 24;
// defaults.global.legend.display = false;


class App extends Component {
  constructor(props,context){
    super(props,context);
    this.state = {
      pollJson: pollData.polls,
      selected: pollData.polls[0]
    }
  }
  render() {
    let pollListCss = {
      display: 'flex',
      flexWrap: 'wrap'
    }
    const {pollJson, selected} = this.state;
    return (
      <div className="App" style={{padding:'10%'}}>
        <Poll votes={158} json={selected} />
        <Card style={{marginTop:'10%'}}>
          <CardBody>
            <Row>
              <Col xs="12" sm="6">
                <PollList />
              </Col>
              <Col  xs="12" sm="6">
                <PollList />
              </Col>
            </Row>
            <Row>
              <Col xs="12" sm="6">
                <PollList />
              </Col>
              <Col  xs="12" sm="6">
                <PollList />
              </Col>
            </Row>
          </CardBody>
        </Card>
      </div>
    );
  }
}

export default App;
