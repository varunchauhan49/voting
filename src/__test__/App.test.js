import { shallow } from 'enzyme';
import App from '../App.js';
import React from 'react';

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});