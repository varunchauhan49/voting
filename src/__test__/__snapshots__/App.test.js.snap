// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Poll
          json={
            Object {
              "answer": Object {
                "options": Array [
                  Object {
                    "id": 1,
                    "label": "Yes",
                  },
                  Object {
                    "id": 2,
                    "label": "No",
                  },
                ],
                "type": "Single",
              },
              "id": 1,
              "publishedDate": 1516605447,
              "title": "Is bitcoin worth the time and money that mining requires?",
            }
          }
          votes={158}
        />,
        <Card
          style={
            Object {
              "marginTop": "10%",
            }
          }
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
            </Row>
            <Row
              tag="div"
            >
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
            </Row>
          </CardBody>
        </Card>,
      ],
      "className": "App",
      "style": Object {
        "padding": "10%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "json": Object {
            "answer": Object {
              "options": Array [
                Object {
                  "id": 1,
                  "label": "Yes",
                },
                Object {
                  "id": 2,
                  "label": "No",
                },
              ],
              "type": "Single",
            },
            "id": 1,
            "publishedDate": 1516605447,
            "title": "Is bitcoin worth the time and money that mining requires?",
          },
          "votes": 158,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CardBody
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
            </Row>
            <Row
              tag="div"
            >
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
              <Col
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <PollList />
              </Col>
            </Row>
          </CardBody>,
          "style": Object {
            "marginTop": "10%",
          },
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>,
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>,
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <PollList />,
                    "sm": "6",
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                    "xs": "12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <PollList />,
                    "sm": "6",
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                    "xs": "12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>,
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <PollList />,
                    "sm": "6",
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                    "xs": "12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <PollList />,
                    "sm": "6",
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                    "xs": "12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Poll
            json={
              Object {
                "answer": Object {
                  "options": Array [
                    Object {
                      "id": 1,
                      "label": "Yes",
                    },
                    Object {
                      "id": 2,
                      "label": "No",
                    },
                  ],
                  "type": "Single",
                },
                "id": 1,
                "publishedDate": 1516605447,
                "title": "Is bitcoin worth the time and money that mining requires?",
              }
            }
            votes={158}
          />,
          <Card
            style={
              Object {
                "marginTop": "10%",
              }
            }
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>
            </CardBody>
          </Card>,
        ],
        "className": "App",
        "style": Object {
          "padding": "10%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "json": Object {
              "answer": Object {
                "options": Array [
                  Object {
                    "id": 1,
                    "label": "Yes",
                  },
                  Object {
                    "id": 2,
                    "label": "No",
                  },
                ],
                "type": "Single",
              },
              "id": 1,
              "publishedDate": 1516605447,
              "title": "Is bitcoin worth the time and money that mining requires?",
            },
            "votes": 158,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardBody
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>
              <Row
                tag="div"
              >
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
                <Col
                  sm="6"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                  xs="12"
                >
                  <PollList />
                </Col>
              </Row>
            </CardBody>,
            "style": Object {
              "marginTop": "10%",
            },
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Row
                  tag="div"
                >
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>
                </Row>,
                <Row
                  tag="div"
                >
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>
                  <Col
                    sm="6"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                    xs="12"
                  >
                    <PollList />
                  </Col>
                </Row>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Col
                      sm="6"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <PollList />
                    </Col>,
                    <Col
                      sm="6"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <PollList />
                    </Col>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <PollList />,
                      "sm": "6",
                      "tag": "div",
                      "widths": Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ],
                      "xs": "12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <PollList />,
                      "sm": "6",
                      "tag": "div",
                      "widths": Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ],
                      "xs": "12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Col
                      sm="6"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <PollList />
                    </Col>,
                    <Col
                      sm="6"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <PollList />
                    </Col>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <PollList />,
                      "sm": "6",
                      "tag": "div",
                      "widths": Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ],
                      "xs": "12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <PollList />,
                      "sm": "6",
                      "tag": "div",
                      "widths": Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ],
                      "xs": "12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
